Synchronized RetrieveBlob() and pre save with promises.

dlemon opened the pull request:
When testing mongoose-fs with my application running mongodb 3.0 on Windows, I noticed that retrieveBlobs  came back too early, so my grid-fs keys in the document were always undefined. The same synchronization issue I observed for a save.

I have synchronized the retrieveBlobs() and pre-save with Promises to resolve this issue. If interested you can pull them in into your master branch. Dependency is shifted to node >=0.12. Promises are now native to node from this version on.

I have modified the JSON package, because I needed to get this fix out to npm (developers are working on a web application that needs to store large files. The developers are located across the world).

If you opt to pull in the change, then please modify the JSON package, and I will remove the fork/npm package.

Thanks!

Commit:
Synchronized RetrieveBlob() and pre save with promises.

Commit:
Added files to npm ignore. Changed dependency node.js from >=0.10 to >=0.12

Commit:
Changed package name and updated README.

Commit:
changed version to 0.2.9

rricard commented on README.md:
```diff
@@ -1,6 +1,7 @@
-# mongoose-fs
+# mongoose-fs-fork
```
If you want it back into master, you have to remove mentions of the fork

rricard commented on lib/mongoose-fs.js:
```diff
@@ -32,77 +32,84 @@ var mongooseFSPlugin = function (schema, options) {
-        next();
-      }
-    };
+                var saved = function (err, fileInfo) {
```
Please configure your editor to use 2 spaces instead of tabulations

rricard commented on package.json:
```diff
@@ -1,15 +1,15 @@
-  "version": "0.2.7",
-  "description": "Mongoose plugin for large attributes storage in GridFS",
-  "author": "Robin Ricard <ricard.robin@gmail.com>",
+  "name": "mongoose-fs-fork",
```
same here, you have to remove the fork references if you need it to be published

rricard commented on package.json:
```diff
@@ -1,15 +1,15 @@
+  "name": "mongoose-fs-fork",
+  "version": "0.2.9",
+  "description": "Fork of depricated Mongoose plugin for large attributes storage in GridFS",
+  "author": "Dennis Lemon <dennis.lemon@sudselekt.com>",
```
please use a list of authors

rricard commented on package.json:
```diff
@@ -1,15 +1,15 @@
   "main": "./index.js",
   "repository": {
     "type": "git",
-    "url": "git@github.com:rricard/mongoose-fs.git"
```
please get back to the original url

rricard commented:
It's ok for me but you'll have to revert the package.json and readme. And please use 2 space indents instead of a tabulation. If you are able to change this, I'll be able to release a new version on npm.

Thank you for your contribution !

dlemon committed:
Update README.md

dlemon committed:
Update package.json

dlemon committed:
Merge pull request #1 from dlemon/dlemon-patch-1

Update README.md

dlemon committed:
Update mongoose-fs.js

dlemon committed:
Update mongoose-fs.js

dlemon committed:
set indentation to two spaces.

dlemon commented:
Hi Robin, 

I've restored README.md to your original file. 
package.json is restored, except that now it is dependend on node >=0.12
indentation is updated  to two spaces in mongoose-fs.js. I've used the Github editor in browser for this.

Super if you pull in this request, will be my first contribution as an open source developer!

Thanks,
Dennis.

rricard commented:
Is there any reason for being forced to use 0.12 ? If it's not necessary (can we polyfill the promises ?), we shouldn't upgrade it. (I unfortunately know codebases stuck in 0.10). I'm sorry to get picky on this, I promise I merge you right after that ! 

Note: you don't have to publish the fork to npm next time: just reference your git fork in the package.json !

dlemon commented:
Hi Robin,

Good question, and you are not being picky. I will have a look how to keep node on >=0.10. Because of work I will come back to that later this week. On your note: I will do that instead, thanks for reminding.

Have a good Sunday!
Dennis.

rricard commented:
Thanks ! Don't worry, you have your time !

Commit:
Node dependency >=0.10

Implemented promises with npm module q

dlemon committed:
fixed indentation

dlemon commented:
Hi Robin,

I have restored the dependency on node >= 0.10. Added a dependency on the npm module q, With this module I have implemented the promises. All tests run fine. This should do the trick.

For a project I also had to downgrade to 0.10 for dependency on kerberos authentication modules. I was suprised that 0.10 indeed has a large installation base, and 0.12 is really too early adopted by me.

Anyways, look at the changes and let me know what you think.

Thanks!

Commit:
node >=0.10 in package.json

Commit:
Merge branch 'master' of https://github.com/dlemon/mongoose-fs

dlemon committed:
check ci build test for node 0.10.x

dlemon commented:
Brr, too much commits here. Learning git by doing. Next time I will fork my master, test it and pull back in. Sorry for all the commit clutter!

dlemon committed:
Update package.json

dlemon commented:
Hi Robin,

I have restored the dependency on node >= 0.10. Added a dependency on the npm module q, With this module I have implemented the promises. All tests run fine. This should do the trick.

For a project I also had to downgrade to 0.10 for dependency on kerberos authentication modules. I was suprised that 0.10 indeed has a large installation base, and 0.12 is really too early adopted by me.

Anyways, look at the changes and let me know what you think.

Thanks!

Commit:
Updated travis.yml node.js 0.10

Commit:
Merge branch 'master' of https://github.com/dlemon/mongoose-fs

dlemon commented:
Hi Robin,

I have restored the dependency on node >= 0.10. Added a dependency on the npm module q, With this module I have implemented the promises. All tests run fine. This should do the trick.

For a project I also had to downgrade to 0.10 for dependency on kerberos authentication modules. I was suprised that 0.10 indeed has a large installation base, and 0.12 is really too early adopted by me.

Anyways, look at the changes and let me know what you think.

Thanks!

rricard commented:
Nice! I'll merge this soon!

On Thu, Mar 26, 2015, 02:09 Dennis Lemon <notifications@github.com> wrote:

> Hi Robin,
>
> I have restored the dependency on node >= 0.10. Added a dependency on the
> npm module q, With this module I have implemented the promises. All tests
> run fine. This should do the trick.
>
> For a project I also had to downgrade to 0.10 for dependency on kerberos
> authentication modules. I was suprised that 0.10 indeed has a large
> installation base, and 0.12 is really too early adopted by me.
>
> Anyways, look at the changes and let me know what you think.
>
> Thanks!
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/rricard/mongoose-fs/pull/6#issuecomment-86274339>.
>

rricard commented:
Hey and for the "commit clutter" I don't mind. Next time you may want to just `git rebase -i master`

rricard commented:
Usually it's better to work on your local repository on your computer

rricard commented on package.json:
```diff
@@ -18,7 +18,8 @@
   "dependencies": {
-    "mongoose": "3.8.x"
+    "mongoose": "3.8.x",
+	"q": "1.2.x"
```
indent: 2 spaces

rricard commented:
Just a style nit to fix and we're good to go ;)

dlemon committed:
Style fix

dlemon commented:
Great. Learned a lot and thank you for your constructive feedback. The style fix is applied!

rricard commented:
Merged :)
